pkg/diagnostics/types/flags.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"github.com/openshift/origin/pkg/cmd/server/start"
+	flag "github.com/spf13/pflag"
+	"strings"
+)
+
+type Flags struct {
+	Diagnostics    List // named diagnostics to run
+	OpenshiftPath  string
+	OscPath        string
+	LogLevel       int
+	Format         string          // format of output - text/json/yaml
+	OpenshiftFlags *flag.FlagSet   // flags that were set on the command run
+	CanCheck       map[Target]bool // components to diagnose - master/node/client
+	MustCheck      Target
+	// for diagnosing "all", enable specifying config file locations
+	ClientConfigPath string
+	MasterConfigPath string
+	NodeConfigPath   string
+	// the master has its own container for flags/options
+	MasterOptions *start.MasterOptions
+	// the node has its own container for flags/options
+	NodeOptions *start.NodeOptions
```

Comment:
This feels gorpy.  Looking at the package, it seems like this set object gets passed to commands even if those commands aren't supposed to check the Node.  What does it look like to have the information required for each command split apart?  Using partial objects makes it harder to reason about what I actually need to set.